source/remoteClient/connectionInfo.py

```diff
@@ -0,0 +1,97 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2015-2025 NV Access Limited, Christopher Toth, Tyler Spivey, Babbage B.V., David Sexton and others.
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+from dataclasses import dataclass
+from enum import Enum
+from urllib.parse import parse_qs, urlencode, urlparse, urlunparse
+
+from . import protocol
+from .protocol import SERVER_PORT, URL_PREFIX
+
+
+class URLParsingError(Exception):
+	"""Raised if it's impossible to parse out the URL"""
+
+
+class ConnectionMode(Enum):
+	MASTER = "master"
+	SLAVE = "slave"
+
+
+class ConnectionState(Enum):
+	CONNECTED = "connected"
+	DISCONNECTED = "disconnected"
+	CONNECTING = "connecting"
+	DISCONNECTING = "disconnecting"
+
+
+@dataclass
+class ConnectionInfo:
+	hostname: str
+	mode: ConnectionMode
+	key: str
+	port: int = SERVER_PORT
+	insecure: bool = False
+
+	def __post_init__(self):
+		self.port = self.port or SERVER_PORT
+		self.mode = ConnectionMode(self.mode)
+
+	@classmethod
+	def fromURL(cls, url):
+		parsedUrl = urlparse(url)
+		parsedQuery = parse_qs(parsedUrl.query)
+		hostname = parsedUrl.hostname
+		port = parsedUrl.port
+		key = parsedQuery.get("key", [""])[0]
+		mode = parsedQuery.get("mode", [""])[0].lower()
+		insecure = parsedQuery.get("insecure", ["false"])[0].lower() == "true"
+		if not hostname:
+			raise URLParsingError("No hostname provided")
+		if not key:
+			raise URLParsingError("No key provided")
+		if not mode:
+			raise URLParsingError("No mode provided")
+		try:
+			ConnectionMode(mode)
+		except ValueError:
+			raise URLParsingError("Invalid mode provided: %r" % mode)
+		return cls(hostname=hostname, mode=mode, key=key, port=port, insecure=insecure)
+
+	def getAddress(self):
+		# Handle IPv6 addresses by adding brackets if needed
+		hostname = f"[{self.hostname}]" if ":" in self.hostname else self.hostname
+		return f"{hostname}:{self.port}"
+
+	def _build_url(self, mode: ConnectionMode):
+		# Build URL components
+		netloc = protocol.hostPortToAddress((self.hostname, self.port))
+		params = {
+			"key": self.key,
+			"mode": mode if isinstance(mode, str) else mode.value,
```

Comment:
I'd just assume this is of the proper type. Using an StrEnum will probably help as well here.
```suggestion
			"mode": mode,
```